master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/AbstractMetaManager.java

```diff
@@ -281,25 +283,28 @@ public void restoreMetaFromFile(File file) throws IOException {
             }
           });
 
-      workers.addAll(
+      Set<WorkerInfo> workerInfoSet =
           snapshotMetaInfo.getWorkersList().stream()
               .map(PbSerDeUtils::fromPbWorkerInfo)
-              .collect(Collectors.toSet())
-              .stream()
-              .map(
+              .collect(Collectors.toSet());
+      scala.collection.Seq<String> workerHostSeq =
+          JavaConverters.asScalaIterator(workerInfoSet.stream().map(WorkerInfo::host).iterator())
+              .toSeq();
+      scala.collection.immutable.Map<String, Node> resolveMap =
+          rackResolver.resolveToMap(workerHostSeq);
+      workers.addAll(
+          workerInfoSet.stream()
+              .peek(
                   workerInfo -> {
                     // Reset worker's network location with current master's configuration.
                     workerInfo.networkLocation_$eq(
-                        rackResolver.resolve(workerInfo.host()).getNetworkLocation());
```

Comment:
When the number of workers is large, the performance of parsing one by one will decrease.

YARN-9332. RackResolver tool should accept multiple hosts
https://issues.apache.org/jira/browse/YARN-9332